databricks/koalas/groupby.py

```diff
@@ -2016,6 +2013,15 @@ class SeriesGroupBy(GroupBy):
     def __init__(self, kser: Series, by: List[Series], as_index: bool = True):
         self._kser = kser
         self._groupkeys = by
+        # TODO: this class resolves the groupkeys and agg_columns always by columns names
+        #   e.g., F.col("..."). This is because of the limitation of `SeriesGroupBy`
+        #   implementation, which reuses the implementation in `GroupBy`.
+        #   `SeriesGroupBy` creates another DataFrame and
+        #   internal IDs of the columns become different. Maybe we should refactor the whole
+        #   class in the future.
+        self._groupkeys_scols = [F.col(name_like_string(s.name)) for s in self._groupkeys]
+        self._agg_columns_scols = [F.col(name_like_string(s.name)) for s in self._agg_columns]
```

Comment:
`F.col(s._internal.data_columns[0])` instead of `F.col(name_like_string(s.name))`?

Reply:
I switched but realised that some tests fail with that change. Seems like there's some cases when internal column names and series names are different:

```python
>>> (ks.range(10).id + 1)._internal.data_columns
['(id + 1)']
>>> (ks.range(10).id + 1).name
'id'
```

Seems we use:

```python
    @property
    def _kdf(self) -> DataFrame:
        series = [self._kser] + [s for s in self._groupkeys if not s._equals(self._kser)]
        return DataFrame(self._kser._kdf._internal.with_new_columns(series))
```

which always aliases via using `column_index`. So, it seems guaranteed to have the internal Spark column names same as `s.name`. I think using `name` here for this workaround is correct.